kpi/views/v2/asset.py

```diff
@@ -487,24 +484,18 @@ def get_serializer_context(self):
             # The serializer will be able to pick what it needs from that dict
             # and narrow down data according to users' permissions.
             queryset = super().get_queryset()
-            asset_content_type = ContentType.objects.get_for_model(Asset)
 
             # 1) Retrieve all asset IDs of current list
+            asset_ids = AssetPagination.get_all_asset_ids_from_queryset(self.filter_queryset(queryset))
 
-            # Micro optimization, cast `asset_ids` as a list to force the query
-            # to be processed right now. Otherwise, because queryset is lazy query
-            # it creates joins on tables when queryset is interpreted which is
-            # slower than running this extra query.
-            asset_ids = list(self.filter_queryset(queryset).
-                             values_list('id', flat=True).distinct())
             # 2) Get object permissions per asset
             object_permissions = ObjectPermission.objects.filter(
                 asset_id__in=asset_ids,
                 deny=False,
             ).select_related(
                 'user', 'permission'
             ).order_by(
-                'user__username', 'permission__codename'
+                'user_id', 'user__username', 'permission__codename'
```

Comment:
`user_id` has been added for this micro-optimization. 
https://github.com/kobotoolbox/kpi/pull/2811/files#diff-8050e58b297cd93312a0b200d09d6637R511. To be jure, first user in the loop is `AnonymousUser`

Reply:
I removed it. We don't need this anymore.